File: common/src/main/scala/se/kth/id2203/link/NetworkHeader.scala
package se.kth.id2203.link

import se.sics.kompics.network.Address
import se.sics.kompics.network.Header
import se.sics.kompics.network.Transport

class NetworkHeader(val src: Address, val dst: Address, val ptc: Transport) extends Header[Address] {

  override def getSource: Address = this.src

  override def getDestination: Address = this.dst

  override def getProtocol: Transport = this.ptc

}

File: common/src/main/java/se/kth/id2203/link/NetworkHeader.java
package se.kth.id2203.link;

import java.io.Serializable;

import se.sics.kompics.network.Address;
import se.sics.kompics.network.Header;
import se.sics.kompics.network.Transport;

public class NetworkHeader implements Serializable, Header<Address> {

    private final static long serialVersionUID = (- 1611726940513460931L);
    public final Address src;
    public final Address dst;
    public final Transport ptc;

    public NetworkHeader(Address src, Address dst, Transport ptc) {
        this.src = src;
        this.dst = dst;
        this.ptc = ptc;
    }

    @Override
    public Address getSource() {
        return src;
    }

    @Override
    public Address getDestination() {
        return dst;
    }

    @Override
    public Transport getProtocol() { return ptc; }

    @Override
    public String toString() { return this.getClass()+"("+src+","+dst+","+ptc+")"; }
}